lang/cs/Org.Apache.REEF.IO.Tests/TestFileDataSet.cs

```diff
@@ -0,0 +1,255 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Org.Apache.REEF.IO.PartitionedData;
+using Org.Apache.REEF.IO.PartitionedData.FileSystem;
+using Org.Apache.REEF.IO.PartitionedData.FileSystem.Parameters;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Tang.Implementations.Tang;
+using Org.Apache.REEF.Tang.Util;
+
+namespace Org.Apache.REEF.IO.Tests
+{
+    /// <summary>
+    /// Tests for Org.Apache.REEF.IO.PartitionedData.FileSystem.
+    /// </summary>
+    [TestClass]
+    public class TestFileDataSet
+    {
+        /// <remarks>
+        /// This test creates IPartitionDataSet with FileSystemPartitionConfiguration module.
+        /// It then instantiates each IPartition using the IConfiguration provided by the IPartitionDescriptor.
+        /// </remarks>
+        [TestMethod]
+        public void TestEvaluatorSide()
+        {
+            var dataSet = TangFactory.GetTang()
+                .NewInjector(FileSystemPartitionConfiguration<byte>.ConfigurationModule
+                    .Set(FileSystemPartitionConfiguration<byte>.FilePaths, "abc;ppp")
+                    .Set(FileSystemPartitionConfiguration<byte>.FilePaths, "def;ooo;iii")
+                    .Set(FileSystemPartitionConfiguration<byte>.FileSystemType, FileSystemType.Local.ToString())
+                    .Set(FileSystemPartitionConfiguration<byte>.FileSerializer, GenericType<ByteSerializer>.Class)
+                    .Build())
+                .GetInstance<IPartitionedDataSet>();
+
+            foreach (var partitionDescriptor in dataSet)
+            {
+                var partition =
+                    TangFactory.GetTang()
+                        .NewInjector(partitionDescriptor.GetPartitionConfiguration())
+                        .GetInstance<IPartition<IEnumerable<byte>>>();
+                Assert.IsNotNull(partition);
+                Assert.IsNotNull(partition.Id);
+            }
+        }
+
+        /// <remarks>
+        /// This test creates IPartitionDataSet using the configuration build directly.
+        /// It sets multiple files in each Partition
+        /// </remarks>
+        [TestMethod]
+        public void TestWithoutConfigurationModule()
```

Comment:
Additional configuration can always be created and merged no matter we use configuration module or not. Configuration module is a better way to control optional and required parameters and make a configuration handy to use. 
For serializer, since I have changed it to pass configuration as a serialized string, that would allow any parameters in the constructor of the serializer.